toolchain/lower/file_context.cpp

```diff
@@ -297,6 +297,10 @@ auto FileContext::BuildFunctionDefinition(SemIR::FunctionId function_id)
       bind_name_id = addr->inner_id;
     }
     auto bind_name = sem_ir().insts().Get(bind_name_id);
+    if (auto pattern = bind_name.TryAs<SemIR::BindingPattern>()) {
+      bind_name_id = pattern->bind_inst_id;
+      bind_name = sem_ir().insts().Get(bind_name_id);
+    }
```

Comment:
The `addr` check above is doing the same thing, why does this look so different?